Neos.Neos/Classes/Fusion/Cache/CacheTag.php

```diff
@@ -51,9 +51,18 @@ final public static function forNodeAggregateFromNode(Node $node): self
         );
     }
 
+    final public static function forNodeAggregateFromNodeWithoutWorkspace(Node $node): self
+    {
+        return self::forNodeAggregate(
+            $node->contentRepositoryId,
+            null,
+            $node->aggregateId
+        );
```

Comment:
...change this to sth like
```suggestion
        return new self(
            'Node_'
            . self::getHashForContentRepositoryId($contentRepositoryId)
            . '_' . $nodeAggregateId->value
        );
```
and we could reduce the amount of repetition, by introducing some private `fromParts` constructor, e.g.:
```php
return self::fromParts(self::NODE_PREFIX, self::hashed($contentRepositoryId), $nodeAggregateId->value);
```

that would also make it easier to change our internal pattern because I think:
`<prefix>_sha1(<workspace>@<cr>)_<aggregateId>` is not safe (because the `NodeAggregateId` could contain underscores for example – not currently, but it is being [requested](https://github.com/neos/neos-development-collection/issues/5111))

`<prefix>|sha1(<workspace>)|sha1(<cr>)|<aggregateId>` would make more sense to me – even though: I don't understand why we randomly(?) use sha1 for 2 of those?

Reply:
The sha1 is to reduce/fix the length of the key.

Reply:
> The sha1 is to reduce/fix the length of the key

It was used to avoid special characters to render the tags inconsistent – but this should no longer be an issue with our restrictive ValueObjects.
Re length:
The max length of a cache tag is **250** characters.
SHA1 hashes are **40** characters long.
workspaceName can be up to **200** characters
cr id only **16** (i.e. hashing the cr ID _increases_ its length)

We _could_ just hash the complete tag, but that would obviously make it impossible to read so I would not suggest to do so.
Instead I'd say (sorry, this is not related to your change, but since we talk about it):
- Reduce the length of `WorkspaceName` dramatically – 200 characters is crazy
- use a delimiter that can' be part of aggregate id, workspace name or cr id
- don't hash the parts at all

> There is also no need to change this, as this is just a visual delimiter, without any meaning

Well, it's not _just_ visual, imagine a node aggregate id of `foo_notAWorkspace` :)

Reply:
Sure, but can we move this into a dedicated issue?

Reply:
Yes, sorry ;)

Reply:
created an issue because the sha1 bugged me while debugging: https://github.com/neos/neos-development-collection/issues/5164